connector/spanmetricsconnectorv2/config.go

```diff
@@ -44,17 +44,44 @@ const (
 
 // Config for the connector
 type Config struct {
-	Spans map[string]MetricInfo `mapstructure:"spans"`
+	Spans []MetricInfo `mapstructure:"spans"`
 }
 
 // MetricInfo for a data type
 type MetricInfo struct {
+	Name        string            `mapstructure:"name"`
 	Description string            `mapstructure:"description"`
 	Attributes  []AttributeConfig `mapstructure:"attributes"`
 	Unit        MetricUnit        `mapstructure:"unit"`
 	Histogram   HistogramConfig   `mapstructure:"histogram"`
 }
 
+// isEqual checks if two metric have a same identity. Identity of a
+// metric is defined by name and attribute.
+func (mi MetricInfo) isEqual(other MetricInfo) bool {
+	if mi.Name != other.Name {
+		return false
+	}
+	if len(mi.Attributes) != len(other.Attributes) {
+		return false
+	}
+	if len(mi.Attributes) == 0 {
+		return true
+	}
+	// Validate attribues equality
+	keyMap := make(map[string]AttributeConfig)
+	for _, attr := range mi.Attributes {
+		keyMap[attr.Key] = attr
+	}
+
+	for _, otherAttr := range other.Attributes {
+		if _, ok := keyMap[otherAttr.Key]; !ok {
+			return false
+		}
+	}
```

Comment:
I see this handles `mi.Attributes` being a subset of `other.Attributes`. But does this handle `other.Attributes` being a strict subset of `mi.Attributes`?